hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -453,7 +454,7 @@ private boolean initializeFromFilesystem(String initializationTime, List<Metadat
               continue;
             }
             partitionName = expressionIndexPartitionsToInit.iterator().next();
-            fileGroupCountAndRecordsPair = initializeExpressionIndexPartition(partitionName, instantTimeForPartition);
+            fileGroupCountAndRecordsPair = initializeExpressionIndexPartition(partitionName, dataTableInstantTime);
```

Comment:
This is the main change.  Note that `BLOOM_FILTERS` partition uses `dataTableInstantTime`/`initializationTime` instead of `instantTimeForPartition` for bloom filter records in the metadata table.

Reply:
Does it affect the correctness issue?